### PR TITLE
Add cron functions

### PR DESCRIPTION
**Pull request by tooptoop4**
query:
```
select cron_to_english('*/12 4-7 15-22 * 1-4') f, 1
union all
select cron_to_english('0 18 15-22 * *') f,2
union all
select cron_last_time('*/12 4-7 15-22 * 1-4') f,3
union all
select cron_last_time('0 18 15-22 * *') f,4
union all
select cron_next_time('*/12 4-7 15-22 * 1-4') f,5
union all
select cron_next_time('0 18 * * *') f,6
union all
select cron_2nd_last_time('*/12 4-7 15-22 * 1-4') f,7
union all
select cron_2nd_last_time('0 18 15-22 * *') f,8
```

result:
```
every 12 minutes every hour between 4 and 7 every day between 15 and 22 every day between Monday and Thursday	1
at 18:00 every day between 15 and 22	2
2021-01-21T06:12Z[Europe/London]	3
2021-01-20T18:00Z[Europe/London]	4
2021-01-21T06:24Z[Europe/London]	5
2021-01-21T18:00Z[Europe/London]	6
2021-01-21T06:00Z[Europe/London]	7
2021-01-19T18:00Z[Europe/London]	8
```

Useful scenario:
SLAs are stored as cron expressions in table1
Actual runtimes of some ETL jobs in table2
Can then evaluate a job's actual runtimes table against the SLAs

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/operator/scalar/StringFunctions.java [hunk 1]
```diff
@@ -913,6 +922,62 @@ public static boolean startsWith(@SqlType("varchar(x)") Slice source, @SqlType("
         }
         return source.compareTo(0, prefix.length(), prefix, 0, prefix.length()) == 0;
     }
+    
+    @Description("Describes a cron expression in english")
```
Comment (Praveen2112) [inline presto-main/src/main/java/io/prestosql/operator/scalar/StringFunctions.java:926]: Can we extract these function in a separate class specific to Cron function ?

File: presto-main/src/main/java/io/prestosql/operator/scalar/StringFunctions.java [hunk 2]
```diff
@@ -913,6 +922,62 @@ public static boolean startsWith(@SqlType("varchar(x)") Slice source, @SqlType("
...
+    @ScalarFunction("cron_last_time")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice cronLast(@SqlType(StandardTypes.VARCHAR) Slice slice)
+    {
+        CronDefinition cronDefinition = CronDefinitionBuilder.instanceDefinitionFor(UNIX);
+        CronParser parser = new CronParser(cronDefinition);
+        String cronexp = slice.toStringUtf8();
+        ZonedDateTime now = ZonedDateTime.now();
+        ExecutionTime executionTime = ExecutionTime.forCron(parser.parse(cronexp));
+        Optional<ZonedDateTime> lastExecution = executionTime.lastExecution(now);
```
Comment (Praveen2112) [inline presto-main/src/main/java/io/prestosql/operator/scalar/StringFunctions.java:950]: What if lastExecution is not present ? Can we return it as null ?

File: presto-main/src/main/java/io/prestosql/operator/scalar/StringFunctions.java [hunk 3]
```diff
@@ -913,6 +922,62 @@ public static boolean startsWith(@SqlType("varchar(x)") Slice source, @SqlType("
...
+    
+    @Description("Describes a cron expression in english")
+    @ScalarFunction("cron_to_english")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice cronEnglish(@SqlType(StandardTypes.VARCHAR) Slice slice)
+    {
+        //change Locale UTC?
+        CronDescriptor descriptor = CronDescriptor.instance(Locale.UK);
+        CronDefinition cronDefinition = CronDefinitionBuilder.instanceDefinitionFor(UNIX);
+        CronParser parser = new CronParser(cronDefinition);
```
Comment (Praveen2112) [inline presto-main/src/main/java/io/prestosql/operator/scalar/StringFunctions.java:934]: Can we share the `CronParser` ?

**Review by Praveen2112 — COMMENTED** [review]
Thanks for working on this. It would be better if we could remove the unrelated commits and have some tests for it

**Comment by bitsondatadev** [comment]
:wave: @tooptoop4 - this PR is inactive and doesn't seem to be under development. If you'd like to continue work on this at any point in the future, feel free to re-open.

**Pull request closed without merging**